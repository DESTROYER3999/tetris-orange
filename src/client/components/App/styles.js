import styled from "styled-components";

export const StyledContainer = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  &::after {
    content: "";
    background: url(assets/tetris-theme.jpg);
    background-size: cover;
    opacity: 0.2;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    position: absolute;
    z-index: -1;
  }

  // background-size: cover;
`;
