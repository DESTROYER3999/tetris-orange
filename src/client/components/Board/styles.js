import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  position: relative;
  align-self: center;
  background-color: rgba(0, 0, 0, 0.3);
  display: flex;
  flex-direction: column;
  height: 86vh;
  width: 43vh;
  border: 3px solid white;
  border-radius: 3px 0px 0px 3px;

  ${props =>
    props.isSpectre &&
    css`
      border: 2px solid white;
      border-radius: 2px;
      height: 30vh;
      width: 15vh;
      background-color: rgba(0, 0, 0, 0.3);
    `};
`;

export const StyledRow = styled.div`
  flex: 1;
  display: flex;
`;
