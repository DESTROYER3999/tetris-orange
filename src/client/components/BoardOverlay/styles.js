import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  align-self: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  border: 3px solid white;
  border-radius: 5px;
  background-color: rgba(0, 0, 0, 0.5);
  position: absolute;
  top: -3px;
  height: 86vh;
  width: 43vh;
  h2 {
    margin-top: 40px;
    align-self: center;
  }
`;

export const StyledGameOverText = styled.h1`
  align-self: center;
  font-size: 5vh;
`;
