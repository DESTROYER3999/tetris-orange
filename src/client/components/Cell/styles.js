import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  flex: 1 10%;
  border: 1px solid black;
  ${props =>
    props.color &&
    !props.isSpectre &&
    css`
      border: 5px outset ${props => props.color};
      background-color: ${props => props.color};
    `};
  ${props =>
    props.isSpectre &&
    css`
      border: 0px;
      ${props =>
        props.color &&
        css`
          background: #ff8a0080;
        `};
    `};
`;

export const StyledLetter = styled.p`
  font-size: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: font-size 0.2;
  &:hover {
    font-size: 30px;
  }
`;
