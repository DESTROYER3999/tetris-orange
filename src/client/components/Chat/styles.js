import styled from "styled-components";

export const StyledMainContainer = styled.div`
  display: flex;
  flex: 4;
  padding: 10px;
  margin: 5px;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 2px;
`;

export const StyledChatContainer = styled.div`
  display: flex;
  flex: 1;
  border: 3px solid white;
  border-radius: 5px;
  height: 0;
  min-height: 100%;
`;

export const StyledLeftContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  background-color: rgba(0, 0, 0, 0.2);
  border-right: 3px solid white;
`;

export const StyledRightContainer = styled.div`
  flex: 4;
  display: flex;
  flex-direction: column;
`;

export const StyledTitleContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 4vh;
  border-bottom: 3px solid white;
`;
export const StyledMessagesContainer = styled.div`
  display: block;
  flex: 1;
  padding: 5px;
  flex-wrap: wrap;
  overflow-y: auto;
  overflow-x: hidden;
  text-overflow: ellipsis;
`;

export const StyledForm = styled.form`
  display: flex;
  padding: 5px;
`;

export const StyledPlayerListContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  align-items: center;
  padding: 5px;
  border-bottom: 5px;
  overflow-y: auto;
  overflow-x: hidden;
  text-overflow: ellipsis;
`;

export const StyledText = styled.div`
  font-family: Verdana;
  font-size: 1.5vh;
  min-width: 100%;
  width: 0;
  & font {
    font-family: Verdana;
    font-weight: 600;
  }
`;

export const StyledInput = styled.input`
  height: 4vh;
  padding: 0.5vh;
  padding-left: 1vh;
  border-radius: 5px;
  border: 2px solid white;
  font-size: 2vh;
  width: 100%;
  background: none;
  outline: none;
  color: white;
  font-family: Verdana;
  font-size: 1.5vh;
  ::placeholder {
    color: white;
  }
`;
