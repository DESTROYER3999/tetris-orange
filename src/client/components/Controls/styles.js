import styled from "styled-components";

export const StyledContainer = styled.div`
  align-self: center;
  display: flex;
  flex-direction: column;
  width: 70%;
  div {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    kbd {
      font-size: 2vh;
      border-radius: 3px;
      padding: 1px 2px 0;
      border: 3px solid white;
    }
    span {
      font-size: 1.5vh;
    }
  }
`;
