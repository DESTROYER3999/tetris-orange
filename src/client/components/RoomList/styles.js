import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
`;

export const StyledList = styled.div`
  height: 100%;
  border: 3px solid white;
  border-radius: 0px 0px 5px 40px;
  overflow-y: auto;
`;

export const StyledListItem = styled.div`
  height: 5%;
  margin-bottom: 1px;
  margin-top: 1px;
  flex-grow: 1;
  display: flex;
  background-color: rgb(255, 255, 255, 0.1);
  align-items: center;
  transition: background-color 0.3s, font-size 0.3s;
  &:hover {
    background-color: rgb(255, 255, 255, 0.2);
  }
  &:hover #content {
    font-size: 1.8vh;
  }
`;

export const StyledListItemContent = styled.div`
  & #content {
    font-size: 1.5vh;
  }
  display: flex;
  justify-content: center;
  ${css`
    flex: ${props => props.flex};
  `};
`;

export const StyledListSubHeader = styled.div`
  & #header {
    font-size: 2vh;
  }
  height: 6%;
  display: flex;
  align-items: center;
  border: 3px solid white;
  border-bottom: 0px;
  border-radius: 5px 5px 0px 0px;
`;

export const StyledListSubHeaderColumn = styled.div`
  display: flex;
  justify-content: center;
  ${css`
    flex: ${props => props.flex};
  `};
`;
