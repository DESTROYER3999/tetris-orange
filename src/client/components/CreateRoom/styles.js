import styled from "styled-components";
import Input from "../common/Input";
import { withStyles } from "@material-ui/core/styles";

export const StyledForm = styled.form`
  align-self: center;
  width: 70%;
  display: flex;
  padding: 1vh;
  border-radius: 4px;
  justify-content: center;
  align-items: center;
  background: rgb(0, 0, 0, 0.2);
`;

export const StyledRoomNameContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const StyledLabel = styled.label`
  padding: 0.5vh;
  font-size: 2vh;
  color: white;
`;

export const StyledButton = styled.button`
  height: 2vh;
  background: none;
  border: 0px;
  outline: none;
  opacity: 0;
  &:enabled {
    opacity: 1;
  }
  &:disabled {
    heigth: 0px;
  }
`;

export const StyledInput = withStyles({
  root: {
    "&": {
      marginTop: 0,
      marginRight: 10
    },
    "& label.Mui-focused": {
      color: "white"
    },
    "$ .MuiFormHelperText-contained p": {
      height: "2vh"
    },
    "& label": {
      color: "white"
    },
    "& input": {
      color: "white"
    },
    "& .MuiFormHelperText-root": {
      backgroundColor: "white",
      fontSize: "1.5vh"
    },
    "& .MuiOutlinedInput-root": {
      "& fieldset": {
        borderColor: "white"
      },
      "&:hover fieldset": {
        borderColor: "white"
      },
      "&.Mui-focused fieldset": {
        borderColor: "white"
      },
      "&.Mui-error fieldset": {
        borderColor: "white"
      }
    }
  }
})(Input);
