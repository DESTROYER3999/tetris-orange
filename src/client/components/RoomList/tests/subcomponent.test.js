import React from "react";
import { shallow, render } from "enzyme";
import toJson from "enzyme-to-json";
import RoomListSub from "../subcomponent";
import { StyledListItem } from "../styles";
import { SOLO, BATTLEROYAL } from "../../../../constants/game";

const props = {
  rooms: [
    { name: "Room1", playersCount: 2, isStarted: true, mode: BATTLEROYAL },
    { name: "Room2", playersCount: 1, isStarted: false, mode: SOLO }
  ],
  playerName: "Player1"
};

describe("<RoomListSub />", () => {
  test("should render without crashing", () => {
    const wrapper = render(<RoomListSub {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should joinRoom when clicking a roomListItem", done => {
    const wrapper = shallow(<RoomListSub {...props} />);

    const item = wrapper.find(StyledListItem);
    expect(window.location.hash).toBe("");
    item.first().simulate("click");
    setTimeout(() => {
      expect(window.location.hash).toBe(
        "#" + props.rooms[0].name + "[" + props.playerName + "]"
      );
      done();
    }, 50);
  });
});
