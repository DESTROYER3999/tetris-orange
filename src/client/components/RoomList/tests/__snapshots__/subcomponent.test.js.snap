// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomListSub /> should render without crashing 1`] = `
<div
  class="sc-bdVaJa mWeMt"
  id="room-list"
>
  <h2>
    Rooms :
  </h2>
  <div
    class="sc-ifAKCX feFkUc"
    id="subheader"
  >
    <div
      class="sc-EHOje kgNXRr"
    >
      <p
        id="header"
      >
        Name
      </p>
    </div>
    <div
      class="sc-EHOje qZCVm"
    >
      <p
        id="header"
      >
        Mode
      </p>
    </div>
    <div
      class="sc-EHOje qZCVm"
    >
      <p
        id="header"
      >
        Players
      </p>
    </div>
    <div
      class="sc-EHOje qZCVm"
    >
      <p
        id="header"
      >
        Status
      </p>
    </div>
  </div>
  <div
    class="sc-bwzfXH kUfuHL"
    id="list"
  >
    <div
      class="sc-htpNat eHEvwL"
      id="list-item"
    >
      <div
        class="sc-bxivhb bKDjvz"
        id="content"
      >
        Room1
      </div>
      <div
        class="sc-bxivhb iuOKpD"
        id="content"
      >
        Battleroyal
      </div>
      <div
        class="sc-bxivhb iuOKpD"
        id="content"
      >
        2/5
      </div>
      <div
        class="sc-bxivhb iuOKpD"
        id="content"
      >
        In game
      </div>
    </div>
    <div
      class="sc-htpNat eHEvwL"
      id="list-item"
    >
      <div
        class="sc-bxivhb bKDjvz"
        id="content"
      >
        Room2
      </div>
      <div
        class="sc-bxivhb iuOKpD"
        id="content"
      >
        Solo
      </div>
      <div
        class="sc-bxivhb iuOKpD"
        id="content"
      >
        1/1
      </div>
      <div
        class="sc-bxivhb iuOKpD"
        id="content"
      >
        In Menu
      </div>
    </div>
  </div>
</div>
`;
